### ISSUE TITLE
inf_MCRM and MCRM weights names discreptancy

### ISSUE THREAD

**Issue by piercus** [qianyu-dlut/MVANet#3]
@qianyu-dlut thanks for this great work, i was having one more question regarding `MCRM` module.

`MCRM` naming is `linear1` and `linear2` [here](https://github.com/finegrain-ai/mvanet/blob/5c29c237bd63870f73c8f253c7e16faccc7534a7/model/MVANet.py#L246-L247)

`inf_MCRM` naming is `linear3` and `linear4` [here](https://github.com/finegrain-ai/mvanet/blob/5c29c237bd63870f73c8f253c7e16faccc7534a7/model/MVANet.py#L303-L304)

In `Model_80.pth`, there are 4 different layers `linear1` / `linear2` / `linear3` / `linear4`
```
dec_blk1.linear1.weight
dec_blk1.linear1.bias
dec_blk1.linear2.weight
dec_blk1.linear2.bias
dec_blk1.linear3.weight
dec_blk1.linear3.bias
dec_blk1.linear4.weight
dec_blk1.linear4.bias
```

and the values are different : 
```python
import torch

pretrained_dict = torch.load("./saved_model/MVANet/Model_80.pth", map_location='cuda')
print('dec_blk1.linear1.weight', torch.sum(pretrained_dict['dec_blk1.linear1.weight']))
print('dec_blk1.linear3.weight', torch.sum(pretrained_dict['dec_blk1.linear3.weight']))
```
outputs
```
dec_blk1.linear1.weight tensor(2.0187, device='cuda:0')
dec_blk1.linear3.weight tensor(-0.5632, device='cuda:0')
```

What is the difference between `linear1` and `linear3` ?

Thanks for your help

**Comment by htrvu** [comment]
The pretrained weights also have some strange keys, e.g. `multifieldcrossatt.attention.5`, `dec_blk1.attention.4`, when the model state dict does not have those keys (multifieldcrossatt only has attention.0 to 4). Was the pretrained weights be trained at a larger model architecture?

**Comment by qianyu-dlut** [comment]
Hi @piercus ! Thank you for bringing this to our attention. There were indeed two additional linear initializations when training the Model_80.th, namely linear1 and linear2, which were not utilized in the subsequent operations. This oversight has led to the situation you've observed.

Furthermore, you are correct regarding the naming convention within the MCRM; linear1 and linear2 should indeed be renamed to linear3 and linear4 to maintain consistency throughout the codebase. We appreciate your keen eye on this matter.

**Comment by qianyu-dlut** [comment]
Hi @htrvu ! Thank you for your careful observation regarding the pretrained weights. You are indeed correct that there were additional modules initialized during the training phase. These modules, however, were not utilized in the forward pass and, as a result, they were omitted from the code for the sake of simplicity and clarity. We apologize for any confusion this may have caused.

**Comment by set-soft** [comment]
I don't want to be rude @qianyu-dlut but this looks like a lot of wrong cut and paste when copying BEN code.
In the original work the layers are layer1 and layer2 indeed, MVANet just renames them to layer5 and layer6, the rest of the code is almost the same.
There are more bogus layers on the pretrained weights. As I load the weights in strict mode (something I strongly recommend you) I had to not only remove sideout (difference between inference and training classes) but also apply this filter:

```python
MVANET_MCLM_BUG = re.compile(r'(multifieldcrossatt\.(linear[12]|attention\.[567]))|'
                             r'(dec_blk\d\.(linear[12]|attention\.[4567]))')
```

Looking at the code I see this is BEN (the first release of the code used to train the BEN_base.pth) changing activation functions and using BatchNorm2d instead of InstanceNorm2d. Can you elaborate on this?

Also, I don't see where is [BEN](https://arxiv.org/abs/2501.06230) cited, what is the reference?

BTW, you should try using the code released when they trained BEN2, is the same network, but the inference was adapted to support videos.

**Comment by piercus** [comment]
@set-soft maybe i missed something here but [BEN](https://github.com/PramaLLC/BEN) is from Jan 2025 while [MVANet](https://github.com/qianyu-dlut/MVANet/commit/03c50197823968a24675d100ab6b76f107b1ec4a) is April 2024

**Comment by set-soft** [comment]
Hi @piercus !

Both are 2024, take a look at the BEN repo. But you are right about which one is older according to the history.

The first commit of this repo is https://github.com/qianyu-dlut/MVANet/commit/03c50197823968a24675d100ab6b76f107b1ec4a dated Apr 26, 2024

And the HuggingFace commit for BEN is https://huggingface.co/PramaLLC/BEN/commit/acd779eb3920d1136b658f8566fa302cbe405f96 dated Nov 11, 2024

So this code looks older, you are right. Thanks for the clarification!

P.S. I added attribution to this work to my copy of the code and marked BEN changes as posterior
